### PR TITLE
Bugfix/ponderluminance d3 d 2283

### PR DESCRIPTION
**Pull request by Kuranes**
- Ponders luminance by solid angle
- reduce light area size after zone light merge to select area light near analytic directional point

### CODE REVIEW

File: ExtractLightsVariance [hunk 1]
```diff
@@ -6,18 +6,20 @@
 
 #include "Math"
 
-struct double2 {
+struct Double2 {
```
Comment (cedricpinson) [inline ExtractLightsVariance]: Why not using Vec2d already in Math ? same with Vec3d

**Comment by cedricpinson** [comment]
@Kuranes  It seems that extractLight is still built but could probably be removed right ? if we use the extractLights we dont need the previous one.

**Comment by Kuranes** [comment]
True, can be removed if no "existing" code conflicts (to be checked with tests done by backend ?)

**Comment by Kuranes** [comment]
Have maths errors on the fast path of the integral image building, on big image + "vibrant" color, as summing give great values, whereas colors themself can be small values too, have to test
- slow path (not accumulative sum in one pass, but each pixel summing whole sum "behind" him)
- reducing size of input to 1024 or 512

**Comment by Kuranes** [comment]
Added Error Detection, tried quick fix for enhancing precision, but need more work...

**Pull request closed without merging**